pkg/kubelet/server/stats/types.go

```diff
@@ -70,6 +72,8 @@ type PodStats struct {
 	// Stats pertaining to volume usage of filesystem resources.
 	// VolumeStats.UsedBytes is the number of bytes used by the Volume
 	VolumeStats []VolumeStats `json:"volume,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+	// User defined metrics that are exposed by containers in the pod. Typically, we expect only one container in the pod to be exposing custom metrics. In the event of multiple containers exposing metrics, they will be combined here.
+	UserDefinedMetrics []UserDefinedMetric `json:"userDefinedMetrics,omitmepty" patchStrategy:"merge" patchMergeKey:"name"`
```

Comment:
hmm, this structure is a little funny... now "sample" is a top-level construct (the whole summary is a single time-stamped sample), it seems a little funny to have multiple UserDefinedMetric samples with different time stamps nested within the structure. How will we decide which `UserDefinedMetricValue`s (samples) to include with the summary?